parquet-column/src/main/java/org/apache/parquet/io/RecordReaderImplementation.java

```diff
@@ -424,15 +428,68 @@ public T read() {
     return record;
   }
 
+  private State getState(MessageType schema) {
+    for (State state : states) {
+        if (schema.containsField(state.fieldPath[0])) {
+          return state;
+        }
+    }
+    throw new IllegalArgumentException("Couldn't find state for schema " + schema);
+  }
+
+  protected void readVectors(ColumnVector[] vectors, MessageType[] columnSchemas, long current, long total) {
```

Comment:
`MessageType` is used for root schemas. I'm not sure why one would ever have an array of them.